azure/iac/KeyVault.rego

```diff
@@ -217,4 +217,356 @@ enablePurgeProtection_metadata := {
+}
+
+source_path[{"keyvault_service_endpoint":metadata}] {
+    resource := input.resources[i]
+    lower(resource.type) == "microsoft.keyvault/vaults"
+    virtualNetworkRule := resource.properties.networkAcls.virtualNetworkRules[j]
+    not virtualNetworkRule.id
+    metadata:= {
+        "resource_path": [["resources",i,"properties","networkAcls","virtualNetworkRules",j,"id"]]
+    }
+}
+
+azure_attribute_absence ["keyvault_service_endpoint"] {
+    resource := input.resources[_]
+    lower(resource.type) == "microsoft.keyvault/vaults"
+    not resource.properties.networkAcls.virtualNetworkRules
+}
+
+source_path[{"keyvault_service_endpoint":metadata}] {
+    resource := input.resources[i]
+    lower(resource.type) == "microsoft.keyvault/vaults"
+    virtualNetworkRule := resource.properties.networkAcls.virtualNetworkRules
+    metadata:= {
+        "resource_path": [["resources",i,"properties","networkAcls","virtualNetworkRules"]]
+    }
+}
+
+azure_issue ["keyvault_service_endpoint"] {
+    resource := input.resources[_]
+    lower(resource.type) == "microsoft.keyvault/vaults"
+    virtualNetworkRule := resource.properties.networkAcls.virtualNetworkRules[_]
+    count(virtualNetworkRule.id) == 0
+}
+
+source_path[{"keyvault_service_endpoint":metadata}] {
+    resource := input.resources[i]
+    lower(resource.type) == "microsoft.keyvault/vaults"
+    virtualNetworkRule := resource.properties.networkAcls.virtualNetworkRules[j]
+    count(virtualNetworkRule.id) == 0
+    metadata:= {
+        "resource_path": [["resources",i,"properties","networkAcls","virtualNetworkRules",j,"id"]]
+    }
+}
+
+keyvault_service_endpoint {
+    lower(input.resources[_].type) == "microsoft.keyvault/vaults"
+    not azure_attribute_absence["keyvault_service_endpoint"]
+    not azure_issue["keyvault_service_endpoint"]
+}
+
+keyvault_service_endpoint = false {
+    azure_attribute_absence["keyvault_service_endpoint"]
+}
+
+keyvault_service_endpoint = false {
+    azure_issue["keyvault_service_endpoint"]
+}
+
+keyvault_service_endpoint_err = "microsoft.keyvault/vaults resoruce property 'networkAcls.virtualNetworkRules' or 'networkAcls.virtualNetworkRules.id' are missing" {
+    azure_attribute_absence["keyvault_service_endpoint"]
+} else = "Service Endpoint disabled for Azure Key Vault" {
+    azure_issue["keyvault_service_endpoint"]
+}
+
+keyvault_service_endpoint_metadata := {
+    "Policy Code": "PR-AZR-ARM-KV-008",
+    "Type": "IaC",
+    "Product": "AZR",
+    "Language": "ARM template",
+    "Policy Title": "Key Vault should use a virtual network service endpoint",
+    "Policy Description": "This policy audits any Key Vault not configured to use a virtual network service endpoint.",
+    "Resource Type": "microsoft.keyvault/vaults",
+    "Policy Help URL": "",
+    "Resource Help URL": "https://docs.microsoft.com/en-us/azure/templates/microsoft.keyvault/vaults"
+}
+
+
+# PR-AZR-ARM-KV-009
+#
+
+default kv_private_endpoint = null
+
```

Comment:
@shahinshirvani for this type of rule, please follow the similar suggestion at https://github.com/prancer-io/prancer-compliance-test/pull/357